### PR TITLE
feat: introduce native functions for Int.ediv / Int.emod

### PR DESCRIPTION
**Pull request by joehendrix**
These still need tests, but I thought I'd upstream so I can use benchmarking and check for build errors.

### CODE REVIEW

**Comment by joehendrix** [comment]
!bench

**Comment by joehendrix** [comment]
!bench

File: src/runtime/mpz.cpp [hunk 1]
```diff
@@ -160,6 +160,43 @@ mpz & mpz::operator*=(unsigned u) { mpz_mul_ui(m_val, m_val, u); return *this; }
...
+}
+
+mpz mpz::emod(mpz const & n, mpz const & d) {
+    mpz r;
+    /* (q,r) = (n/d, n%d) */
+    mpz_tdiv_r(r.m_val, n.m_val, d.m_val);
+    /* if (r < 0) */
+    if (mpz_sgn(r.m_val) < 0) {
+        if (mpz_sgn(d.m_val) > 0) {
+            /* r = r - d. */
```
Comment (joehendrix) [inline src/runtime/mpz.cpp:190]:
```suggestion
            /* r = r + d. */
```

File: src/runtime/mpz.cpp [hunk 2]
```diff
@@ -699,6 +736,80 @@ mpz & mpz::operator%=(mpz const & o) {
...
+    q1.ensure_capacity(q_sz);
+    r1.ensure_capacity(r_sz);
+    mpn_div(m_digits, m_size,
+            digits, sz,
+            q1.begin(), r1.begin());
+    set(q_sz, q1.begin());
+    m_sign = !is_zero() && m_sign != sign;
+    return *this;
+}
+
```
Comment (joehendrix) [inline src/runtime/mpz.cpp:762]:
```suggestion
```

File: src/runtime/mpz.h [hunk 3]
```diff
@@ -245,6 +245,14 @@ class mpz {
 
     friend mpz operator%(mpz a, mpz const & b) { return a %= b; }
 
+    static mpz ediv(mpz const & n, mpz const & d);
+    static mpz ediv(int n, mpz const & d) { return ediv(mpz(n), d); }
+    static mpz ediv(mpz const& n, int d) { return ediv(n, mpz(d)); }
+
+    static mpz emod(mpz const & n, mpz  const & d);
+    static mpz emod(int n, mpz const &  d) { return emod(mpz(n), d); }
+    static mpz emod(mpz const &  n, int d) { return emod(n, mpz(d)); };
```
Comment (joehendrix) [inline src/runtime/mpz.h:254]:
```suggestion
    static mpz emod(mpz const & n, mpz const & d);
    static mpz emod(int n, mpz const & d) { return emod(mpz(n), d); }
    static mpz emod(mpz const & n, int d) { return emod(n, mpz(d)); };
```

**Comment by joehendrix** [comment]
I've added some tests, so I think this is in a reasonable state now.

**Comment by digama0** [comment]
This looks like a good use case for `test_extern`

**Comment by joehendrix** [comment]
What is `test_extern`?

**Comment by digama0** [comment]
@joehendrix https://github.com/leanprover/lean4/blob/master/RELEASES.md?plain=1#L379-L386

**Pull request merged**